version: "3.8"

networks:
  cfa-database_cfa-network:
    external: true

services:
  cfa-backend:
    container_name: cfa-api
    build:
      context: ./
      dockerfile: ./app/Dockerfile
      args:
        - UID=${UID:-1000}
        - GID=${GID:-1000}
    volumes:
      - ./app/api:/var/www/html:delegated
    restart: unless-stopped
    networks:
      cfa-database_cfa-network:
        ipv4_address: 172.31.0.22

  web-server:
    container_name: cfa-nginx-server
    build:
      context: ./
      dockerfile: ./nginx/Dockerfile
      args:
        - UID=${UID:-1000}
        - GID=${GID:-1000}
    ports:
      - 31002:80
    volumes:
      - ./app/api:/var/www/html:delegated
      - ./nginx/default.conf:/etc/nginx/conf.d/default.conf
    depends_on:
      - cfa-backend
    restart: unless-stopped
    networks:
      cfa-database_cfa-network:
        ipv4_address: 172.31.0.23